### PR TITLE
Nginx version to support HTTP2

### PR DESCRIPTION
**Pull request by emord**
@ctsims @shubham1g5 I think it makes sense to spin up a separate proxy for staging since the instructions included removing nginx on the system. THoughts?

https://github.com/JeffreyBLewis/WebVoteView/wiki/Instructions-for-upgrading-NGINX-to-allow-HTTP2-support-on-Ubuntu-14.04

### CODE REVIEW

**Comment by shubham1g5** [comment]
@emord Yeah. Lets push this code to staging. We can then run all our integration tests against that and see if anything breaks because of this.

**Comment by emord** [comment]
This is on softlayer swiss and staging. will deploy to prod, nic and enikshay on monday-ish

Once this is deployed we can remove the line to remove the old nginx repository and start using http2

**Comment by emord** [comment]
Just deployed nginx 1.12.2 to all environments. Going to set up another PR to enable HTTP2 to deploy tomorrow (keeping it to one change at a time).

Couple of commits in here that came from differences on ICDS https://github.com/dimagi/commcarehq-ansible/pull/872/commits/5bb29a49475b905025f2d32a3d93fd9b87382ffd https://github.com/dimagi/commcarehq-ansible/pull/872/commits/2dbc0ce884c97356435cf7a46b95a1b6ff3090fb Guessing @calellowitz or @snopoke have the most context on that

also as noted on #devops swiss is currently on 12.04, so I was unable to deploy to that. I added the ubuntu verison as a variable so that I could change what repo to use on each environment

File: ansible/vars/dev/dev_public.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ COMMTRACK_SITE_HOST: '{{ groups.proxy.0 }}'
 NO_WWW_SITE_HOST: '{{ groups.proxy.0 }}'
 J2ME_SITE_HOST: 'j2me.{{ groups.proxy.0 }}'
 internal_domain_name: 'localdomain'
+ubuntu_version: "14.04"
```
Comment (snopoke) [inline ansible/vars/dev/dev_public.yml]: I think you can use `ansible_distribution_version` for this

Comment (emord) [reply ansible/vars/dev/dev_public.yml]: Thanks I didn't know about that variable. much cleaner now

**Pull request merged**